maint.php

```diff
@@ -819,57 +845,235 @@ function applyFilter() {
 		strURL += '&associated=' + $('#associated').is(':checked');
 		strURL += '&filter=' + $('#filter').val();
 		strURL += '&header=false';
+		strURL += '&site_id=' + $('#site_id').val();
+		strURL += '&poller_id=' + $('#poller_id').val();
+		strURL += '&location=' + $('#location').val();
 		loadPageNoHeader(strURL);
 	}
 
 	function clearFilter() {
 		strURL = 'maint.php?tab=hosts&action=edit&id=<?php print get_request_var('id');?>&clear=true&header=false'
 		loadPageNoHeader(strURL);
 	}
+
+	$(function() {
+		$('#rows, #site_id, #poller_id, #location, #host_template_id, #filter, #associated').change(function() {
+			applyFilter();
+		});
+
+		$('#clear').click(function() {
+			clearFilter();
+		});
+
+		$('#form_devices').submit(function(event) {
+			event.preventDefault();
+			applyFilter();
+		});
+	});
 	</script>
 	<?php
 
-	html_start_box(__('Associated Devices %s', htmlspecialchars($header_label), 'maint'), '100%', '', '3', 'center', '');
+	html_start_box(__('Associated Devices %s', $header_label), '100%', '', '3', 'center', '');
```

Comment:
You seem to have removed the escaping around the header label.  I believe this could result in an XSS issue.

You have also removed the 'maint' (plugin) override to the language processor.

Reply:
I mistook the escaping in get_header_label().  I tested <script> but did not notice any problems.  The escaping is added back.

Reading through the function __() code, I now understand the last parameter somewhat.  It is added back.